src/registry/ecr.go

```diff
@@ -147,18 +161,45 @@ func (r *RegistryScan) WaitForScanFindings(ctx context.Context, digestInfo Image
 	maxAttemptDelay := 15 * time.Second
 	maxTotalDelay := 3 * time.Minute
 
-	err := waiter.Wait(ctx, &ecr.DescribeImageScanFindingsInput{
-		RegistryId:     &digestInfo.RegistryID,
-		RepositoryName: &digestInfo.Name,
-		ImageId: &types.ImageIdentifier{
-			ImageDigest: &digestInfo.Digest,
+	err := waiter.Wait(ctx,
+		&ecr.DescribeImageScanFindingsInput{
+			RegistryId:     &digestInfo.RegistryID,
+			RepositoryName: &digestInfo.Name,
+			ImageId: &types.ImageIdentifier{
+				ImageDigest: &digestInfo.Digest,
+			},
+			MaxResults: aws.Int32(1), // reduce the size of the return payload when waiting for the completion state
 		},
-		MaxResults: aws.Int32(1), // reduce the size of the return payload when waiting for the completion state
-	}, maxTotalDelay, func(opts *ecr.ImageScanCompleteWaiterOptions) {
-		opts.LogWaitAttempts = true
-		opts.MinDelay = minAttemptDelay
-		opts.MaxDelay = maxAttemptDelay
-	})
+		maxTotalDelay,
+		func(opts *ecr.ImageScanCompleteWaiterOptions) {
+			opts.LogWaitAttempts = true
+			opts.MinDelay = minAttemptDelay
+			opts.MaxDelay = maxAttemptDelay
+		},
+		func(opts *ecr.ImageScanCompleteWaiterOptions) {
+			/*
+				We must copy this function outside the closure to avoid an infinite loop
+				If we copy it inside the closure the compiler will assign it to a pointer
+				to itself
+			*/
+			defaultRetryableFunc := opts.Retryable
+			customRetryable := func(ctx context.Context, params *ecr.DescribeImageScanFindingsInput,
+				output *ecr.DescribeImageScanFindingsOutput, err error) (bool, error) {
+
+				if err != nil && strings.Contains(err.Error(), "AccessDeniedException") {
```

Comment:
Added this case to return immediately if we get an AccessDenied error. Otherwise we falsely get the "exceeded max wait time for ImageScanComplete waiter" error.